src/coreclr/vm/comsynchronizable.cpp

```diff
@@ -557,31 +523,8 @@ FCIMPL1(void, ThreadNative::Initialize, ThreadBaseObject* pThisUNSAFE)
 }
 FCIMPLEND
 
-
-// Set whether or not this is a background thread.
-FCIMPL2(void, ThreadNative::SetBackground, ThreadBaseObject* pThisUNSAFE, CLR_BOOL isBackground)
-{
-    FCALL_CONTRACT;
-
-    if (pThisUNSAFE==NULL)
-        FCThrowResVoid(kNullReferenceException, W("NullReference_This"));
-
-    // validate the thread
-    Thread  *thread = pThisUNSAFE->GetInternal();
-
-    if (ThreadIsDead(thread))
-        FCThrowResVoid(kThreadStateException, W("ThreadState_Dead_State"));
-
-    HELPER_METHOD_FRAME_BEGIN_0();
-
-    thread->SetBackground(isBackground);
-
-    HELPER_METHOD_FRAME_END();
-}
-FCIMPLEND
-
 // Return whether or not this is a background thread.
-FCIMPL1(FC_BOOL_RET, ThreadNative::IsBackground, ThreadBaseObject* pThisUNSAFE)
+FCIMPL1(FC_BOOL_RET, ThreadNative::GetIsBackground, ThreadBaseObject* pThisUNSAFE)
 {
     FCALL_CONTRACT;
 
```

Comment:
(for future PRs)

We will need to get rid of the `FCThrowRes` too. I think the best way to do it - without regressing performance - is to duplicate some of the state bits on the managed thread object, so that the IsDead check can be done in C#.